### ISSUE TITLE
Add volume create option to specify thick provisioning for performance

### ISSUE THREAD

**Issue by corrieb** [vmware/vic#6322]
Hal Rosenburg's recent performance testing has shown that write performance is impacted by our use of thin provisioning of volumes. 

See http://rickardnobel.se/eager-thick-vs-lazy-thick-disk-performance/ also.

This absolutely makes sense as the default, given that we don't want to be efficient about datastore usage, but we should provide the option for thick provisioning if you want maximum throughput in production.

From a policy standpoint, it might be better to characterize it as "lazy or eager". 

Implementation would simply be `docker volume create --opt Provision=[thin|lazythick|eagerthick] myVol`

**Comment by hickeng** [comment]
I have WIP code for this rolled up in a disk-opts branch however that branch currently includes work towards raw disk support as well (no filesystem).

We should split out my changes and address this directly, with raw disk support being added in a separate issue. I'm also going to make this depend on #7397 as the re-organisation makes these changes less scattered.
